### PR TITLE
UIE-563 Updated readme.md

### PR DESCRIPTION
**Pull request by stepek**
#### What has changed in the code ####
Add info about config json and ui schema

#### Reasons for making this change ####
Missing some information in readme

### Checklist

* [ ] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated docs if needed
* [ ] **I'm adding a new feature**
  - [ ] I've added unit test for feature
  - [ ] I've ran lint

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,14 +48,65 @@ Setup development environment
 ```
 
 Config file
-------------------------------------
+----------------------------------------------------------------------------------------------------------------------
 `config.json` file should be located in the same directory as all source files. Sample configuration is included in `config.json.sample` file.
 
 Possible fields in config file:
-
+- `authUrl` - URL to keystone
```
Comment (sealionkat) [inline README.md]: `Keystone` :)

File: README.md [hunk 2]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+- `gohan` - gohan connection config, must contain object with following fields:
```
Comment (sealionkat) [inline README.md]: Probably better: `Gohan connection config, must be an object ....`

File: README.md [hunk 3]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+  - `schema` - path to jsonschema
```
Comment (sealionkat) [inline README.md]: `JSON Schema`

File: README.md [hunk 4]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+  - `url` - URL to gohan server
```
Comment (sealionkat) [inline README.md]: `Gohan`

File: README.md [hunk 5]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+- `sidebar` - array of custom sidebar buttons, must contain objects with following fields:
+  - `path`
+  - `title` 
+  - `id`
+- `sidebarFavorites` - array of button id's to shows in the upper part of the sidebar
```
Comment (sealionkat) [inline README.md]: `buttons' IDs` or `button's IDs`?
or 
`array of links to views, must contain resources' IDs`

File: README.md [hunk 6]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+- `sidebarCategories` - array of object to grouping buttons in sidebar menu categories, must contain objects with following fields:
```
Comment (sealionkat) [inline README.md]: `array of objects including information about views' categories, must contain objects with...`

File: README.md [hunk 7]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+  - `name` - display name of category
+  - `id` - id of category
```
Comment (sealionkat) [inline README.md]: `ID`

File: README.md [hunk 8]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+  - `items` - array of button id's to shows in the category
```
Comment (sealionkat) [inline README.md]: as earlier `array of links to views...`

File: README.md [hunk 9]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
 - `sidebarChildResources` - array of schema IDs of child resources which should be visible in sidebar menu
 - `routes` - config array for routing, must contain objects with following fields:
   - `path` - URL path to view
   - `viewClass` - name of React component related to the view (in `src/routes/componentList.js`)
   - `name` - name of route
   - `exact` - match path exactly, default `true`
+- `errorMessages` - map of error messages
+  - `tokenExpire` - token expired message
+- `addingRelationDialog` - array of ????
+- `pageLimit` - limit of the page size
```
Comment (sealionkat) [inline README.md]: or `limit of displayed items in page`?

File: README.md [hunk 10]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+- `loginRequestTimeout` - time before token expiration, when token renew dialog should be shown in ms
```
Comment (sealionkat) [inline README.md]: `time left (in ms) to token expiration (when token renew dialog should be shown)

File: README.md [hunk 11]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+- `polling` - flag to enable polling
```
Comment (sealionkat) [inline README.md]: we can add here possible values, ex. `must be true or false` etc.

File: README.md [hunk 12]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+- `pollingInterval` - interval of polling
```
Comment (sealionkat) [inline README.md]: `interval (in ms)`

File: README.md [hunk 13]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+- `storagePrefix` - prefix to store token information (local and session storage)
```
Comment (sealionkat) [inline README.md]: `(local- and sessionStorage)`

File: README.md [hunk 14]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+- `tableDefaultSortKey` - default sort key
```
Comment (sealionkat) [inline README.md]: we can add here example of possible values

File: README.md [hunk 15]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+- `tableDefaultSortOrder` - default sort order
```
Comment (sealionkat) [inline README.md]: we should add here example of possible values

File: README.md [hunk 16]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+- `title`- title of web page
+- `followableRelations` - flat to enable followable resource by reference
```
Comment (sealionkat) [inline README.md]: `flag` ;)
`flag to enable following resource by reference`

File: README.md [hunk 17]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+- `useKeystoneDomain` - flag to enable use domain feature in keystone v3
```
Comment (sealionkat) [inline README.md]: `flag to enable using domain feature in Keystone v3`
How about adding a link to Keystone doc?

File: README.md [hunk 18]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+- `selectDomainFromHost` - regex to select domain from host (keystone V3)
```
Comment (sealionkat) [inline README.md]: `Keystone v3`

File: README.md [hunk 19]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+- `domainName` - default domain name
+
+
+UI schema
+----------------------------------------------------------------------------------------------------------------------
+`locales/en-us/uiSchema.json` file defined specific behaviors in dialogs like conditions hiding and showing fields
```
Comment (sealionkat) [inline README.md]: `file containg the definition of specific UI behaviors in dialogs like hiding and showing fields`

File: README.md [hunk 20]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+
+Json file should contains array of objects with following fields:
```
Comment (sealionkat) [inline README.md]: `The JSON file should contain ...`

File: README.md [hunk 21]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+ - `id` - id of json schema resource
```
Comment (sealionkat) [inline README.md]: `ID of JSON Schema ...`

File: README.md [hunk 22]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+ - `title` - (optional) - title of resource
+ - `properties` - (optional) - properties customization, objects with following fields:
```
Comment (sealionkat) [inline README.md]: Is it object or array of objects?

File: README.md [hunk 23]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+       - `ui:labelTemplate` - (optional) - label custom template for select field
+       - `ui:requiredResource` - (optional) - custom require collection for select field
+       - `ui:query` - (optional) - query to fetch additional data for select field
+ - `logic` - (optional) - custom dialog logic
+    - `[fieldName]` - (optional) - field name
+       - `[fieldName]` - (optional) - field name (if parent is Object) if is array should be `items` 
+          - `[value]` - (optional) - specified value of field
+             - `hide` - array of fields to hide
+
+Other method of customization using uischema can be find in [react-jsonschema-form documentation](https://github.com/rjsf-team/react-jsonschema-form/tree/v0.51.0#form-customization)
```
Comment (sealionkat) [inline README.md]: `Other methods of customization using UI schema can be find ...`

File: README.md [hunk 24]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+    - `[fieldName]` - (optional) - customization of single field, objects with following fields:
+       - `ui:widget` - (optional) -  alternative widget
+       - `ui:field` - (optional) - alternative field
+       - `ui:step` - (optional) - specified step in number input field
+       - `ui:labelTemplate` - (optional) - label custom template for select field
+       - `ui:requiredResource` - (optional) - custom require collection for select field
+       - `ui:query` - (optional) - query to fetch additional data for select field
+ - `logic` - (optional) - custom dialog logic
+    - `[fieldName]` - (optional) - field name
+       - `[fieldName]` - (optional) - field name (if parent is Object) if is array should be `items`
```
Comment (sealionkat) [inline README.md]: `field name (if parent is Object), when array, `[fieldName]` should be `items`

File: README.md (hunk 14 above)
Comment (stepek) [reply README.md]: we don't know possible values :)

File: README.md (hunk 22 above)
Comment (stepek) [reply README.md]: object

File: README.md (hunk 14 above)
Comment (sealionkat) [reply README.md]: Oh, I see.

File: README.md (hunk 22 above)
Comment (sealionkat) [reply README.md]: so `object with...` ;)

File: README.md [hunk 25]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+  - `items` - array of links to views in the category
 - `sidebarChildResources` - array of schema IDs of child resources which should be visible in sidebar menu
 - `routes` - config array for routing, must contain objects with following fields:
   - `path` - URL path to view
   - `viewClass` - name of React component related to the view (in `src/routes/componentList.js`)
   - `name` - name of route
   - `exact` - match path exactly, default `true`
+- `errorMessages` - map of error messages
+  - `tokenExpire` - token expired message
+- `addingRelationDialog` - array of ????
```
Comment (michal-damiecki) [inline README.md]: As far as I know this field isn't used in any of our UIs so IMO we can remove it from README, what do you think?

File: README.md [hunk 26]
```diff
@@ -48,14 +48,65 @@ Setup development environment
...
+ - `id` - ID of JSON Schema resource
+ - `title` - (optional) - title of resource
+ - `properties` - (optional) - properties customization, objects with following fields:
+    - `[fieldName]` - (optional) - customization of single field, objects with following fields:
+       - `ui:widget` - (optional) -  alternative widget
+       - `ui:field` - (optional) - alternative field
+       - `ui:step` - (optional) - specified step in number input field
+       - `ui:labelTemplate` - (optional) - label custom template for select field
+       - `ui:requiredResource` - (optional) - custom require collection for select field
+       - `ui:query` - (optional) - query to fetch additional data for select field
```
Comment (michal-damiecki) [inline README.md]: I'm not sure if that's all the fields we use to customize properties.
Please check `ui:order`, `ui:step`.

Comment (stepek) [reply README.md]: `ui:step` is described on 102 line,
`ui:order` is described in react jsonschema form documentation

**Pull request merged**